x-pack/plugin/eql/src/test/java/org/elasticsearch/xpack/eql/analysis/VerifierTests.java

```diff
@@ -58,8 +81,280 @@ public void testMisspelledColumnWithMultipleOptions() {
         assertEquals("1:11: Unknown column [pib], did you mean any of [pid, ppid]?", error("foo where pib == 1"));
     }
 
+    public void testPipesUnsupported() {
+        assertEquals("1:20: Pipes are not supported", errorParsing("process where true | head 6"));
+    }
 
-    private static Map<String, EsField> loadEqlMapping(String name) {
-        return TypesTests.loadMapping(name);
+    public void testProcessRelationshipsUnsupported() {
+        assertEquals("2:7: Process relationships are not supported",
+                errorParsing("process where opcode=1 and process_name == \"csrss.exe\"\n" +
+                        "  and descendant of [file where file_name == \"csrss.exe\" and opcode=0]"));
+        assertEquals("2:7: Process relationships are not supported",
+                errorParsing("process where process_name=\"svchost.exe\"\n" +
+                        "  and child of [file where file_name=\"svchost.exe\" and opcode=0]"));
+    }
+
+    public void testSequencesUnsupported() {
+        assertEquals("1:1: Sequence is not supported", errorParsing("sequence\n" +
+                "  [process where serial_event_id = 1]\n" +
+                "  [process where serial_event_id = 2]"));
+    }
+
+    public void testJoinUnsupported() {
+        assertEquals("1:1: Join is not supported", errorParsing("join by user_name\n" +
+                "  [process where opcode in (1,3) and process_name=\"smss.exe\"]\n" +
+                "  [process where opcode in (1,3) and process_name == \"python.exe\"]"));
+    }
+
+    // Some functions fail with "Unsupported" message at the parse stage
+    public void testArrayFunctionsUnsupported() {
+        assertEquals("1:16: Unsupported function [arrayContains]",
+                errorParsing("registry where arrayContains(bytes_written_string_list, 'En')"));
+        assertEquals("1:16: Unsupported function [arraySearch]",
+                errorParsing("registry where arraySearch(bytes_written_string_list, a, a == 'en-us')"));
+        assertEquals("1:16: Unsupported function [arrayCount]",
+                errorParsing("registry where arrayCount(bytes_written_string_list, s, s == '*-us') == 1"));
+    }
+
+    // Some functions fail with "Unknown" message at the parse stage
+    // TODO (@costin): Should we make the error messaging consistent?
```

Comment:
@costin, I left this TODO comment, need your feedback on consistency for error messaging.